aesara/tensor/random/type.py

```diff
@@ -70,6 +70,13 @@ def is_valid_value(a, strict):
             if state_key.shape == (624,) and state_key.dtype == np.uint32:
                 return True
 
+        # Workaround for supporting RandomState in Numba backend
+        if isinstance(a, tuple) and a[0] == "numba_internal_state_arr":
+            # The second element of tuple should be an array representing
+            # numba's internal state
+            if a[1].shape == (624,) and a[1].dtype == np.uint32:
+                return True
```

Comment:
Can we do this without making Numba-specific changes to generic types?

Reply:
The only way I could find to do this without making Numba-specific changes to generic types is that if we make `RandomState` an acceptable datatype in Numba. 

It doesn't necessarily need to hold all the information of the `RandomState` just be a 'fake' placeholder type just to get `RandomState` objects accepted in Numba and whenever it needs to be reconstructed as a `RandomState` (to get it out of the Numba function) we can do that using Numba's internal state array.

Reply:
Yeah, let's add a type for it to Numba; that sounds great.